### PR TITLE
Update MeanAbsoluteError loss reduction

### PR DESCRIPTION
**Pull request by divyashreepathihalli**
Update MeanAbsoluteError loss reduction to be consistent with MeanAbsoluteError metrics reduction behavior.

Fixes https://github.com/keras-team/keras/issues/19740

### CODE REVIEW

File: keras/src/losses/losses.py [hunk 1]
```diff
@@ -99,7 +99,7 @@ class MeanAbsoluteError(LossFunctionWrapper):
 
     def __init__(
         self,
-        reduction="sum_over_batch_size",
+        reduction="mean_absolute_error",
```
Comment (fchollet) [inline keras/src/losses/losses.py:102]: ?

Comment (divyashreepathihalli) [reply keras/src/losses/losses.py:102]: @fchollet 
The reduction function for `keras.metrics.MeanAbsoluteError` is `mean_absolute_error` and the reduction function for `keras.losses.MeanAbsoluteError` is `sum_over_batch_size`.
The user here https://github.com/keras-team/keras/issues/19740 faces an issue when using sample weights for her training. the loss computes a different value from the associated metric during training.
More details here - https://github.com/keras-team/keras/issues/19740

Comment (divyashreepathihalli) [reply keras/src/losses/losses.py:102]: i just realized `mean_absolute_error` is not one of the options, I can add that.
